### ISSUE TITLE
numeric fields exported as text by default since v4 to excel

### ISSUE THREAD

**Issue by tobhv** [django-import-export/django-import-export#1842]
Hello,
not sure whether this is a bug / feature.
In v3 exporting numeric fields (floats, decimal fields) used to give numeric fields in excel when exporting.
in version 4 by default numeric fields are displayed as text.
in case i add a field definition to the resource class the issue dissappears:

```
class SensorResource(resources.ModelResource):
    float_factor = Field(attribute='float_factor', widget=NumberWidget())
    class Meta:
        model = Meter
```

Is that behavior intentional ?

**Comment by matthewhegarty** [comment]
Thanks for raising.  

> Is that behavior intentional ?

Yes, in v3 the widget `render()` method returned a mixture of strings and native types for various widgets.  This caused issues depending on the file type, so we standardised on returning strings.  I guess the ideal would be to export either strings or native types depending on the file format. i.e. csv would require strings, xlsx would require native types.  This would be an enhancement for a future release.

There is the option to export to export the native type, documented [here](https://django-import-export.readthedocs.io/en/latest/advanced_usage.html#modify-render-return-type).

I will update the release notes to clarify the change in behaviour.

**Comment by tobhv** [comment]
thx a bunch for pointing to the documentation. i can confirm that works. however some notes.:
1) fields that are part of a foreign key widget do not get string coercion by default.
2) duration field gets string coercion applied by default.
3) when i use the duration field in a foreign field i cannot enforce string coercion.
example with 'sensor_last' being a foreign field:

```
class MeterResource(resources.ModelResource):
    correction_factor = Field(attribute='correction_factor',
                              widget=NumberWidget(coerce_to_string=False))
    last_measurement = Field(attribute='last_measurement',
                             widget=DurationWidget())

    class Meta:
        model = Meter


class MeterAdmin(ImportExportModelAdmin):
    resource_classes = [MeterResource]


class MeasurementResource(resources.ModelResource):
    time = Field(attribute='time',
                 widget=DateTimeWidget())
    # enforced string coercion works:
    sensor_cf = Field(attribute='sensor',
                      widget=ForeignKeyWidget(Meter,
                                              field='correction_factor',
                                              coerce_to_string=True)
                      )
    # enforced string coercion does not work:

    sensor_last = Field(attribute='sensor',
                        widget=ForeignKeyWidget(Meter,
                                                field='last_measurement',
                                                coerce_to_string=True)
                        )
    value = Field(attribute='value', widget=DecimalWidget())

    class Meta:
        model = Measurement
```

**Comment by tobhv** [comment]
Well, as (also) documented: string coercion can be enforced using dehydrate.
```
    def dehydrate_sensor_last(self, measurement):
        try:
            last_measurement = measurement.sensor.last_measurement
            if last_measurement:
                return str(last_measurement)
        except ObjectDoesNotExist:
            return
```
so. in case excel cannot handle the natively dehydrate / custom method needs to be applied ? this is also the case for datetime objects as far as I can see.

**Comment by matthewhegarty** [comment]
Thanks for looking into this.  It looks like we might have some gaps which need address.

**Comment by matthewhegarty** [comment]
Note that some of the issues regarding export of native types have been addressed in #1939

**Comment by matthewhegarty** [comment]
> fields that are part of a foreign key widget do not get string coercion by default.

There's no easy way around this at present, the best way to format the output is to create a `ForeignKeyWidget` subclass.  i have updated the documentation to clarify this.

> duration field gets string coercion applied by default.

This was addressed in #1939.
